pynestml/utils/global_processing.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+#
+# global_processing.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+from collections import defaultdict
+
+import copy
+
+from pynestml.codegeneration.printers.nestml_printer import NESTMLPrinter
+from pynestml.codegeneration.printers.constant_printer import ConstantPrinter
+from pynestml.codegeneration.printers.ode_toolbox_expression_printer import ODEToolboxExpressionPrinter
+from pynestml.codegeneration.printers.ode_toolbox_function_call_printer import ODEToolboxFunctionCallPrinter
+from pynestml.codegeneration.printers.ode_toolbox_variable_printer import ODEToolboxVariablePrinter
+
+from pynestml.codegeneration.printers.sympy_simple_expression_printer import SympySimpleExpressionPrinter
+from pynestml.meta_model.ast_expression import ASTExpression
+from pynestml.meta_model.ast_model import ASTModel
+from pynestml.meta_model.ast_simple_expression import ASTSimpleExpression
+from pynestml.utils.ast_global_information_collector import ASTGlobalInformationCollector
+from pynestml.utils.ast_utils import ASTUtils
+
+from odetoolbox import analysis
+
+
+class GlobalProcessing:
+    """Manages the collection of basic information necesary for all types of mechanisms and uses the
```

Comment:
To all new files in the utils directory: could you add a note to the docstrings saying that these are for the compartmental code generator?